scipy/stats/stats.py

```diff
@@ -3565,6 +3568,159 @@ def count_rank_tie(ranks):
+
+    Specifying a rank array is meaningful only if you have in mind an
+    external criterion of importance. If, as it usually happens, you do
+    not have in mind a specific rank, the weighted tau is defined by
+    averaging the values obtained using the decreasing lexicographical
+    rank by (x,y) and by (y,x). This is the behavior with default
+    parameters.
+
+    Note that if you are computing the weighted tau on arrays of ranks,
+    rather than of scores (i.e., a larger value implies a lower rank) you
+    must negate the ranks, so that elements of higher rank are associated
+    with a larger value.
+
+    Parameters
+    ----------
+    x, y : array_like
+        Arrays of rankings, of the same shape. If arrays are not 1-D, they will
+        be flattened to 1-D.
+    rank: array_like of ints or False or True, optional
+        A nonnegative rank assigned to each element. If it is None, the
+        decreasing lexicographical rank by (x,y) will be used: elements of
+        higher rank will be those with larger x-values, using y-values to
+        break ties (in particular, swapping x and y will give a different
+        result). If you use False, the element indices will be used
+        directly as ranks. The default is True, in which case this
+        function returns the average of the values obtained using the
+        decreasing lexicographical rank by (x,y) and by (y,x).
+    weigher : function, optional
+        The weigher function. Must map nonnegative integers (zero representing
+        the most important element) to a nonnegative weight. The default is
+        hyperbolic weighing, that is, x -> 1/(x+1).
+    additive : boolean, optional
+        If true, the weight of an exchange is computed by adding the
+        weights of the ranks of the exchanged elements; otherwise, the weights
+        are multiplied. The default is True.
+
+    Returns
+    -------
+    correlation : float
+       The weighted tau correlation index.
+    pvalue : float
+       Presently np.nan, as the null statistics for the weighted tau is
+       unknown (even in the additive hyperbolic case).
+
+    See also
+    --------
+    kendalltau : Calculates Kendall's tau
+    spearmanr : Calculates a Spearman rank-order correlation coefficient.
+    theilslopes : Computes the Theil-Sen estimator for a set of points (x, y).
+
+    References
+    ----------
+    .. [1] Sebastiano Vigna "A weighted correlation index for rankings with ties".
+           In Proceedings of the 24th international conference on World Wide Web,
+           pages 1166-1176. ACM, 2015.
+    .. [2] W.R. Knight, "A Computer Method for Calculating Kendall's Tau with
+           Ungrouped Data", Journal of the American Statistical Association,
+           Vol. 61, No. 314, Part 1, pp. 436-439, 1966.
+
+    Notes
+    -----
+    This function uses an O(n log n), mergesort-based algorithm [1]_ that is a
+    weighted extension of Knight's algorithm for Kendall's tau [2]_.
+
```

Comment:
I'd still explain the Shieh version here including citation, and explain that this implementation based on [1] is a generalized version.